core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java

```diff
@@ -554,6 +555,66 @@ public void testDropNamespace() {
         "is not empty. 1 tables exist.", () -> catalog.dropNamespace(tbl4.namespace()));
   }
 
+  @Test
+  public void testCreateNamespace() {
+    Namespace testNamespace = Namespace.of("testDb", "ns1", "ns2");
+    // Test with null metadata
+    AssertHelpers.assertThrows("Cannot create a namespace with null or empty metadata", IllegalArgumentException.class,
+            () -> catalog.createNamespace(testNamespace, null));
```

Comment:
Can you make a separate test, `testCreateNamespaceFailsWithoutNamespaceProperties()`?

That way, there would be something in the code that self-documents that we are intentional about needing namespace properties.

You could then even do something like the following
```java
AssertHelpers.assertThrows(
    "JDBC catalog cannot create a namespace without adding namespace properties as the properties are used as an existence check for RDBMs that don't support an atomic compare and swap",
    ...
```